ThirdPartyModules/MageWorld/RewardPoints.php

```diff
@@ -0,0 +1,175 @@
+                ) {
+                    $rewardPoints = $this->mwRewardPointsModelCustomer->create()->load($quote->getCustomerId())->getMwRewardPoint();
+                    $amount = abs($this->mwRewardPointsHelperData->exchangePointsToMoneys($rewardPoints, $storeCode));
+                } else {
+                    $amount = $quote->getMwRewardpointDiscount();
+                }
+
+                $roundedAmount = CurrencyUtils::toMinor($amount, $currencyCode);
+                $discounts[] = [
+                    'description'       => 'Reward Points',
+                    'amount'            => $roundedAmount,
+                    'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_STORE_CREDIT,
+                    'discount_type'     => $this->discountHelper->getBoltDiscountType('by_fixed'), // For v1/discounts.code.apply and v2/cart.update
+                    'type'              => $this->discountHelper->getBoltDiscountType('by_fixed'), // For v1/merchant/order
+                ];
+
+                $diff -= CurrencyUtils::toMinorWithoutRounding($amount, $currencyCode) - $roundedAmount;
+                $totalAmount -= $roundedAmount;
+            }
+        } catch (\Exception $e) {
+            $this->bugsnagHelper->notifyException($e);
+        } finally {        
+            return [$discounts, $totalAmount, $diff];
+        }
+    }
+    
+    /**
+     * Add js to the cart page to refresh Bolt cart when the applied reward points changes.
+     *
+     * @param string $result
+     *
+     * @return string
+     */
+    public function getAdditionalJS($result)
+    {
+        try {
+            $result .= 'var mwRewardPointsAjaxData = "";
+            $(document).ajaxSuccess(function( event, xhr, settings ) {
+                if (settings.url.indexOf("rewardpointspost") > -1 && mwRewardPointsAjaxData != settings.data) {
+                    $(document).trigger("bolt:createOrder");
+                    mwRewardPointsAjaxData = settings.data;
+                }
+            });';
+        } catch (\Exception $e) {
+            $this->bugsnagHelper->notifyException($e);
+        } finally {        
+            return $result;
+        }
+    }
+    
+    /**
+     * Update reward points to the quote.
+     *
+     * @param MW\RewardPoints\Helper\Data $mwRewardPointsHelperData
+     * @param MW\RewardPoints\Model\CustomerFactory $mwRewardPointsModelCustomer
+     * @param Quote $quote
+     *
+     */
+    public function beforePrepareQuote($mwRewardPointsHelperData,
+                                     $mwRewardPointsModelCustomer,
+                                     $quote)
+    {
+        $this->mwRewardPointsHelperData = $mwRewardPointsHelperData;
+        $this->mwRewardPointsModelCustomer = $mwRewardPointsModelCustomer;
+        
+        try {
+            if ($quote->getMwRewardpoint()) {
+                $currencyCode = $quote->getQuoteCurrencyCode();
```

Comment:
@dumega I'm not seeing $currencyCode used anywhere in this class. Should we remove it?

Reply:
removed